### PR TITLE
Enable preloading and remove node-hc

### PR DESCRIPTION
**Pull request by sjanuary**

### CODE REVIEW

**Comment by mhdawson** [comment]
I think you need to validate that Bluemix does not use this to integrate with appmetrics and that none of their documentation refers to it.

File: README.md [hunk 1]
```diff
@@ -133,9 +101,16 @@ The call to `appmetrics.monitor()` starts the data collection agent, making the
 
 You should start your application using the `node` command as usual (**not** `node-hc`).
 
-You must call `require('appmetrics');` *before* the require statements for any npm modules you want to monitor. Appmetrics must be initialized first so that it can instrument modules for monitoring as they are loaded. If this is a problem due to the structure of your application you can require the module on the node command line with -r to make sure it is pre-loaded:
+You must call `require('appmetrics');` *before* the require statements for any npm modules you want to monitor. Appmetrics must be initialized first so that it can instrument modules for monitoring as they are loaded. If this is a problem due to the structure of your application you can require the module on the node command line with -r or --require to make sure it is pre-loaded:
```
Comment (gibfahn) [inline README.md]: I'd also mention `export NODE_OPTIONS="require appmetrics/start"`, as that works with docker images and `npm start` (which node-hc does not).

**Comment by AndyRWatson** [comment]
I am happy to approve this on the basis that we clearly document the current/recommended process and its validated for all deployment options.

File: README.md (hunk 1 above)
Comment (sjanuary) [reply README.md]: @gibfahn I've updated the README to also mention NODE_OPTIONS and I've changed 'deprecated' to 'removed' as I think deprecated usually means the function is still available.

**Comment by tobespc** [comment]
To the best of my knowledge, bluemix does not make used of node_hc.  All the apps that use us have us as a dependency and require us in the code.   This is one of the reasons we will move to a new major version.  Its not possible to find out if anyone is using node_hc still but at least updating the major version will give them warning something as changed.

File: README.md [hunk 2]
```diff
@@ -112,14 +81,17 @@ The following options are specific to appmetrics:
...
 
 ```sh
-$ node-hc app.js
+$ node --require appmetrics/start app.js
 ```
+to preload and start appmetrics, or use the NODE_OPTIONS environment variable:
 
-The purpose of this mode of operation is to provide monitoring of the application without requiring any changes to the application code. The data is sent to the Health Center Eclipse IDE client.
+```sh
+$ export NODE_OPTIONS="require appmetrics/start"
```
Comment (gibfahn) [inline README.md]: I think this needs to be ` $ export NODE_OPTIONS="--require appmetrics/start"`, the idea being that you pass the same string to `NODE_OPTIONS` that you'd pass as a command line flag.

File: README.md [hunk 3]
```diff
@@ -133,9 +105,20 @@ The call to `appmetrics.monitor()` starts the data collection agent, making the
...
+```sh
+$ node --require appmetrics myapp.js
+```
+or
+```sh
+$ node --require appmetrics/start myapp.js
+```
+or
+```sh
+$ export NODE_OPTIONS="require appmetrics/start"
```
Comment (gibfahn) [inline README.md]: Same as above, also does this all need to be repeated?

File: README.md [hunk 4]
```diff
@@ -133,9 +105,20 @@ The call to `appmetrics.monitor()` starts the data collection agent, making the
...
+```
+Inlcuding /start means that appmetrics monitoring will also be started. If you don't include /start on the command line appmetrics will be loaded and ready to begin monitoring, but has not yet started.  Call `appmetrics.monitor()` or `appmetrics.start()` in your code to begin monitoring.
```
Comment (gibfahn) [inline README.md]: Nit:

`Inlcuding` -> `Including`

`but has not yet started` -> `but not yet started` (will be ... has sounds weird to me)

File: README.md (hunk 2 above)
Comment (sjanuary) [reply README.md]: thanks

File: README.md (hunk 3 above)
Comment (sjanuary) [reply README.md]: I've consolidated the two sections now

File: README.md [hunk 5]
```diff
@@ -112,16 +81,19 @@ The following options are specific to appmetrics:
...
-### The `node-hc` command
-If you [globally installed](#installation) this module with npm, you can use the `node-hc` command to run your application instead of the `node` command. This will run your application as it would normally under node (including any node options) but additionally load and start `appmetrics`.
+### Preloading appmetrics
+In previous versions appmetrics came with an executable, `node-hc`, which could be used instead of the `node` command to run your application and load and start appmetrics. This has been removed in version 4.0.0, instead you can use:
 
 ```sh
-$ node-hc app.js
+$ node --require appmetrics/start app.js
 ```
+to preload and start appmetrics, or in Node.js 8.0.0 or later, use the NODE_OPTIONS environment variable:
```
Comment (gibfahn) [inline README.md]: It's also in `6.12.0` (and onwards), so maybe

`Node.js 8.0.0, 6.12.0, or later`

or something

Comment (sjanuary) [reply README.md]: done

**Comment by stalleyj** [comment]
Merging the code, goodbye node-hc

**Pull request merged**